### PR TITLE
net-analyzer/nethogs: added capabilities

### PR DESCRIPTION
**Pull request by Alessandro-Barbieri**
Package-Manager: Portage-3.0.14, Repoman-3.0.2
Signed-off-by: Alessandro Barbieri <lssndrbarbieri@gmail.com>

### CODE REVIEW

File: net-analyzer/nethogs/nethogs-0.8.6-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+KEYWORDS="~amd64 ~arm ~ia64 ~x86"
+DOCS=( DESIGN README.decpcap.txt README.md )
+
+RDEPEND="
+	net-libs/libpcap
+	sys-libs/ncurses:0=
+"
+DEPEND="
+	${RDEPEND}
+	virtual/pkgconfig
```
Comment (thesamesam) [inline net-analyzer/nethogs/nethogs-0.8.6-r1.ebuild:23]: BDEPEND

File: net-analyzer/nethogs/nethogs-0.8.6-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+inherit fcaps toolchain-funcs
+
+DESCRIPTION="A small 'net top' tool, grouping bandwidth by process"
+HOMEPAGE="https://github.com/raboof/nethogs"
+SRC_URI="https://github.com/raboof/nethogs/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~ia64 ~x86"
+DOCS=( DESIGN README.decpcap.txt README.md )
```
Comment (thesamesam) [inline net-analyzer/nethogs/nethogs-0.8.6-r1.ebuild:15]: New line?

**Review by thesamesam — CHANGES_REQUESTED** [review]
We need an option for suid still given people can't always use caps, like we discussed

**Comment by Alessandro-Barbieri** [comment]
Isn't suid a security hazard? The ebuild didn't had it in the first place. Also I have no sympathy for people stuck in the past.

**Comment by thesamesam** [comment]
I meant in general with your PRs to remove it. And it's not about "stuck in the past" -- some filesystems can't support it, and so on. I just meant to make accepting your PRs for these easier. I agree with supporting caps where we can.

File: net-analyzer/nethogs/nethogs-0.8.6-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	emake NCURSES_LIBS="$( $(tc-getPKG_CONFIG) --libs ncurses )" "${PN}"
+}
+
+src_install() {
+	emake DESTDIR="${ED}" PREFIX="/usr" install
+	einstalldocs
+}
+
+pkg_postinst() {
+	ecaps cap_net_admin,cap_net_raw "${EROOT}/usr/bin/nethogs"
```
Comment (thesamesam) [inline net-analyzer/nethogs/nethogs-0.8.6-r1.ebuild:37]: - ecaps doesn't exist?
- Isn't it at /usr/sbin/nethogs, not /usr/bin/nethogs?
- We haven't actually used the eclass here

Comment (thesamesam) [reply net-analyzer/nethogs/nethogs-0.8.6-r1.ebuild:37]: Fixed in https://github.com/gentoo/gentoo/commit/6db0315d786c499a9526ce7ee71d3229fea8fd26

**Pull request closed without merging**